runtime/auth/aws-signing-crt/common/src/aws/smithy/kotlin/runtime/auth/awssigning/crt/CrtAwsSigner.kt

```diff
@@ -47,6 +47,14 @@ public object CrtAwsSigner : AwsSigner {
         val crtResult = CrtSigner.signChunk(chunkBody, prevSignature, crtConfig)
         return AwsSigningResult(Unit, crtResult.signature)
     }
+
+    override suspend fun signChunkTrailer(
+        trailingHeaders: ByteArray,
+        finalChunkSignature: ByteArray,
+        config: AwsSigningConfig,
+    ): AwsSigningResult<Unit> {
+        TODO("Not yet implemented")
```

Comment:
I think we can implement this already. We may need to wire up everything in `aws-crt-kotlin` to `aws-crt-java`. We should ensure that whatever interface we define we can actually implement...

Reply:
Yeah I believe I could implement this right now. I took a look at the [aws-crt-java function and it takes a List\<Header\> as input](https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/auth/signing/AwsSigner.java#L116), so I may need to do some deserialization / serialization, but otherwise it should be fine.

Reply:
Implemented in the latest commits here and opened a new PR on the aws-crt-kotlin repo: https://github.com/awslabs/aws-crt-kotlin/pull/60